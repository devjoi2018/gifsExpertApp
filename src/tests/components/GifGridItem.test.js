import { GifGridItem } from "../../components/GifGridItem";
import {shallow} from "enzyme";

describe('Pruebas sobre <GifsGridItem />', () => {

    const title = "titulo";
    const url = "https://www.localhost/algo.jpg";
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

   test('Debe de mostrar correctamente el componente', () => {

       expect(wrapper).toMatchSnapshot();

   });

   test('Debe de tener un parrafo que muestre el titulo', () => {

        expect(wrapper.find('p').text().trim()).toBe(title);
       
   });

   test('Debe de tener la imagen igual a la url y un alt igual al de los props', () => {

        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
       
   });

   test('Debe de tener animate__fadeIn', () => {

        const div = wrapper.find('div');
        expect(div.hasClass('animate__fadeIn')).toBe(true);
       
   });
   
   
   
    
});
