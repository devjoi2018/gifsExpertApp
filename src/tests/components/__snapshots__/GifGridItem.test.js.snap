// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifsGridItem /> Debe de mostrar correctamente el componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="titulo"
    src="https://www.localhost/algo.jpg"
  />
  <p>
    titulo
  </p>
</div>
`;
